kloppy/infra/serializers/tracking/epts/meta_data.py

```diff
@@ -57,21 +69,39 @@ def _load_periods(global_config_elm, frame_rate: int) -> List[Period]:
     return periods
 
 
-def _load_players(players_elm, team_map: Dict[str, Team]) -> List[Player]:
+def _load_players(players_elm, team: Team) -> List[Player]:
     return [
         Player(
-            team=team_map[player_elm.attrib["teamId"]],
+            team=team,
             jersey_no=str(player_elm.find("ShirtNumber")),
             player_id=player_elm.attrib["id"],
             name=str(player_elm.find("Name")),
+            first_name="",
+            last_name="",
+            position=_load_position_data(
+                player_elm.find("ProviderPlayerParameters")
+            ),
             attributes=_load_provider_parameters(
-                players_elm.find("ProviderPlayerParameters")
+                player_elm.find("ProviderPlayerParameters")
             ),
         )
         for player_elm in players_elm.iterchildren(tag="Player")
+        if player_elm.attrib["teamId"] == team.team_id
     ]
 
 
+def _load_position_data(parent_elm) -> Position:
+    player_provider_parameters = _load_provider_parameters(parent_elm)
```

Comment:
The same function _load_provider_parameters get's called twice, on inside _load_position_data and then to set the Player attributes. Not sure that's the proper way to do it, but couldn't think of a better way to do it while still kipping the list comprehension structure.

Reply:
@koenvo just to be sure that you saw this, any feedback here or it's ok as it?

Reply:
Good point! Besides calling the function twice, the ‘Player.attributes’ property should not contain the position data. But I don’t know a clean way to split it right now. If you can just add a todo / note it’s fine, or if you figure at a nice way it’s even better :-)